src/modules/keyboard/SetKeyboardLayoutJob.cpp

```diff
@@ -324,9 +324,36 @@ SetKeyboardLayoutJob::writeDefaultKeyboardData( const QString& defaultKeyboardPa
               "# Consult the keyboard(5) manual page.\n\n";
 
     stream << "XKBMODEL=\"" << m_model << "\"\n";
-    stream << "XKBLAYOUT=\"" << m_layout << "\"\n";
-    stream << "XKBVARIANT=\"" << m_variant << "\"\n";
-    stream << "XKBOPTIONS=\"\"\n\n";
+    if ( m_additionalLayoutInfo.additionalLayout.isEmpty() )
```

Comment:
Perhaps something along these lines:
```
auto removeEmpties = [](QStringList && l) { l.removeAll(QString()); return l; };
const QStringList layouts(removeEmpties({m_additionalLayoutInfo.additionalLayout, m_layout});
const QStringList variants(removeEmpties({m_additionalLayoutInfo.additionalVariant, m_variant});
```
Er, no, that's yucky. Well, you could bind-by-reference `m_additionalLayoutInfo` and pass in two strings, returning s1,s2 or only s2 depending. And the options string is
```
const QString options = m_additionalLayoutInfo.isEmpty() ? QString() : M-additionalLayoutInfo.groupSwitcher();
```
Then you can keep one set of layout-output lines, outputting only the calculated strings.